### DISCUSSION TITLE
user.conf syntax query re www to non www redirect

### DISCUSSION THREAD

**Discussion by Pezzab** [EasyEngine/easyengine#1678] [Support Requests]
Hello,

I'm attemping to redirect a www.domain to non www.

So far the advice I have found suggests putting it in user.conf.

If I wrap the instruction in a server directive and reload nginx - using `ee site restart site.com --nginx` I get an error.

If I leave out the server directive there is no error but the redirect doesn't work:

```
  server_name www.site.com;
  return 301 $scheme://site.com$request_uri;
```

The redirect should work instantly if configured correctly yes?

**Comment by scollonp** [comment]
Have a look in `/var/lib/docker/volumes/global-nginx-proxy_confd/_data/`. Every site you create should have a redirect in here. If not, add one in the format `mysite.com-redirect.conf`. The contents should be something like this:

```
server {
        listen  80;
        server_name  www.mysite.com;
        return  301 https://mysite.com$request_uri;
}
server {
	listen  443;
	ssl_protocols TLSv1.2 TLSv1.3;
	ssl_ciphers 'ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-SHA384:ECDHE-RSA-AES256-SHA384:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA256';
	ssl_prefer_server_ciphers on;
	ssl_session_timeout 5m;
	ssl_session_cache shared:SSL:50m;
	ssl_session_tickets off;
	ssl_certificate /etc/nginx/certs/mysite.com.crt;
	ssl_certificate_key /etc/nginx/certs/mysite.com.key;
	server_name  www.mysite.com;
	return  301 https://mysite.com$request_uri;
}
```

Then restart your site.

**Comment by tmsbrasil** [comment]
I used this command to solve it **ee site update example.com --add-alias-domains='www.example.com'**

**Comment by scollonp** [comment]
Awesome.

On Wed, 30 Mar 2022, at 5:30 PM, tmsbrasil wrote:
> 
> 
> I used this command to solve it *ee site update example.com --add-alias-domains='www.example.com'*
> 
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/EasyEngine/easyengine/discussions/1678#discussioncomment-2471487>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/ADZJBR3VI5DIHPPXTQRMJYTVCR6UFANCNFSM5QI44QOA>.
> You are receiving this because you commented.Message ID: ***@***.***>
> 

Paul Scollon

altFIRE Technologies - Professional Website Design and Hosting Services for businesses in County Louth
Web: https://www.altfire.ie  | Social: @altFIRE
